pkg/build/builder/source.go

```diff
@@ -252,7 +252,16 @@ func extractGitSource(gitClient GitClient, gitSource *api.GitBuildSource, revisi
 		}
 
 		if err := gitClient.Checkout(dir, commit); err != nil {
-			return true, err
+			//TODO - do we want to parse the error string for the text that accompanies a clone / checkout of a PR ref, such as:   pathspec 'refs/pull/1/head' did not match any file(s) known to git
```

Comment:
i think this case is infrequent enough that it's not worth introducing the fragility that would come w/ that optimization.